app/src/main/java/com/google/android/stardroid/source/proto/ProtobufAstronomicalSource.java

```diff
@@ -76,7 +76,7 @@ public ProtobufAstronomicalSource(AstronomicalSourceProto proto, Resources resou
     if (names == null) {
       names = new ArrayList<String>(proto.getNameIdsCount());
       for (int id : proto.getNameIdsList()) {
-        names.add(resources.getString(id));
+        names.add("TestElement");
```

Comment:
Why this change?

Reply:
It can be seen that further iterating through the implementation ```resource``` variable we find that it is an empty collection declared [here](https://github.com/sky-map-team/stardroid/blob/4ec2f8327e4d2ee99493d238bea9c9cb73520b56/app/src/main/java/com/google/android/stardroid/source/proto/SourceProto.java#L2282). Also errors were popping up when building without this fix. Please correct me if I am wrong.

Reply:
Without diving into the code again (it's been a while), I believe that this is where the names are extracted from the protocol buffer data that's bundled with the app.  If you replace this with a hard-coded string I believe it will remove all astronomical names from the app, so the change should be reverted.

Reply:
@barbeau here's the Logcat if the change is reverted
``
2019-03-21 20:33:33.156 19958-20073/? E/AndroidRuntime: FATAL EXCEPTION: pool-1-thread-1
    Process: com.google.android.stardroid, PID: 19958
    android.content.res.Resources$NotFoundException: String resource ID #0x7f0d01e1
        at android.content.res.Resources.getText(Resources.java:351)
        at android.content.res.MiuiResources.getText(MiuiResources.java:97)
        at android.content.res.Resources.getString(Resources.java:445)
        at com.google.android.stardroid.source.proto.ProtobufAstronomicalSource.getLabels(ProtobufAstronomicalSource.java:112)
        at com.google.android.stardroid.layers.AbstractSourceLayer.initialize(AbstractSourceLayer.java:74)
        at com.google.android.stardroid.layers.AbstractFileBasedLayer.access$201(AbstractFileBasedLayer.java:46)
        at com.google.android.stardroid.layers.AbstractFileBasedLayer$1.run(AbstractFileBasedLayer.java:65)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
        at java.lang.Thread.run(Thread.java:764)
``
What may had happened that Resources would have been removed after the Application was made open-source.

Reply:
Have you run the `build_skymap.sh` script?

https://github.com/sky-map-team/stardroid/tree/master/tools

Reply:
Oh I am really I didn't understood the workflow. Should I make this more clear by providing a README.md PR on how to successfully run the app or make a script which integrates the building of  ``build_skymap.sh`` before running the app?

Reply:
@jaydeetay would need to weigh in, but in general it would be great to have `build_skymap.sh` integrated as part of the primary app Gradle build process, so when loaded into Android Studio everything would magically work. But it would need to be done carefully to ensure that all stages aren't repeated on every debug build, and some only occur on first execution or when the data changes.

Reply:
@barbeau Could you please guide me why Travis CI build is failing?

Reply:
Looks like the emulator startup isn't working - I'm guessing it's the same long-running problem I've seen here:
* https://github.com/OneBusAway/onebusaway-android/issues/720
* https://github.com/googlemaps/android-maps-utils/issues/371

I haven't revisited this myself in a while, but it looks like there are other projects now running emulator-based tests on Travis - see https://github.com/OneBusAway/onebusaway-android/issues/720#issuecomment-464410751

Reply:
So do we need to change the ``.travis.yml`` file to add support for higher API emulators?

Reply:
Thanks Utkarsh for your fixes, and Sean for answering.  Yes, the current flow is pretty hideous, but it is documented in tools/README.  The only reason it hasn't been integrated into gradle is I'd much rather make it go away entirely and revamp the way we store the data.  As for the Travis CI build, this has been a frustration for a while.